src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -221,6 +222,9 @@ public function handle(AddProductToOrderCommand $command)
      */
     private function updateOrderCartRules(Order $order, ?OrderInvoice $invoice, array $cartCartRules)
     {
+        CartRule::resetStaticCache();
```

Comment:
I'm surprised that you need to fix this cache issue in the Add handler for a bug that happens in the Update tests 😅
But since you said the bug only occured in the behat tests I guess it's because the cache is set at first when the product is added, still I wonder if this should be applied in the update handlers as well to be sure all the data are always valid